content/_code-samples/tx-serialization/serialize.py

```diff
@@ -0,0 +1,462 @@
+        # low 4 bits is the field code
+        combined_code = (type_code << 4) | field_code
+        return uint8_to_bytes(combined_code)
+    elif type_code >= 16 and field_code < 16:
+        # first 4 bits are zeroes
+        # next 4 bits is field code
+        # next byte is type code
+        byte1 = uint8_to_bytes(field_code)
+        byte2 = uint8_to_bytes(type_code)
+        return b''.join( (byte1, byte2) )
+    elif type_code < 16 and field_code >= 16:
+        # first 4 bits is type code
+        # next 4 bits are zeroes
+        # next byte is field code
+        byte1 = uint8_to_bytes(type_code << 4)
+        byte2 = uint8_to_bytes(field_code)
+        return b''.join( (byte1, byte2) )
+    else: # both are >= 16
+        # first byte is all zeroes
+        # second byte is type
+        # third byte is field code
+        byte2 = uint8_to_bytes(type_code)
+        byte3 = uint8_to_bytes(field_code)
+        return b''.join( (bytes(1), byte2, byte3) )
+
+def vl_encode(vl_contents):
+    """
+    Helper function for length-prefixed fields including Blob types
+    and some AccountID types.
+
+    Encodes arbitrary binary data with a length prefix. The length of the prefix
+    is 1-3 bytes depending on the length of the contents:
+
+    Content length <= 192 bytes: prefix is 1 byte
+    192 bytes < Content length <= 12480 bytes: prefix is 2 bytes
+    12480 bytes < Content length <= 918744 bytes: prefix is 3 bytes
+    """
+
+    vl_len = len(vl_contents)
+    if vl_len <= 192:
+        len_byte = vl_len.to_bytes(1, byteorder="big", signed=False)
+        return b''.join( (len_byte, vl_contents) )
+    elif vl_len <= 12480:
+        vl_len -= 193
+        byte1 = ((vl_len >> 8) + 193).to_bytes(1, byteorder="big", signed=False)
+        byte2 = (vl_len & 0xff).to_bytes(1, byteorder="big", signed=False)
+        return b''.join( (byte1, byte2, vl_contents) )
+    elif vl_len <= 918744:
+        vl_len -= 12481
+        byte1 = (241 + (vl_len >> 16)).to_bytes(1, byteorder="big", signed=False)
+        byte2 = ((vl_len >> 8) & 0xff).to_bytes(1, byteorder="big", signed=False)
+        byte3 = (vl_len & 0xff).to_bytes(1, byteorder="big", signed=False)
+        return b''.join( (byte1, byte2, byte3, vl_contents) )
+
+    raise ValueError("VariableLength field must be <= 918744 bytes long")
+
+
+# Individual field type serialization routines ---------------------------------
+
+def accountid_to_bytes(address):
+    """
+    Serialize an AccountID field type. These are length-prefixed.
+
+    Some fields contain nested non-length-prefixed AccountIDs directly; those
+    call decode_address() instead of this function.
+    """
+    return vl_encode(decode_address(address))
+
+def amount_to_bytes(a):
+    """
+    Serializes an "Amount" type, which can be either XRP or an issued currency:
+    - XRP: 64 bits; 0, followed by 1 ("is positive"), followed by 62 bit UInt amount
+    - Issued Currency: 64 bits of amount, followed by
```

Comment:
Looks like the comment is incomplete. Probably should read "followed by currency and issuer account"